src/runtime.cc

```diff
@@ -102,13 +102,19 @@ namespace detail {
 		MPI_Comm_size(MPI_COMM_WORLD, &world_size);
 		m_num_nodes = world_size;
 
+		m_cfg = std::make_unique<config>(argc, argv);
+		if(m_cfg->is_dry_run()) {
+			if(m_num_nodes != 1) throw std::runtime_error("Cannot perform a dry run with more than one rank.\n");
```

Comment:
That error message might be confusing, it can be read as "The only allowed parameter value for CELERITY_DRY_RUN_NODES is 1". Maybe go with "Celerity must be launched with a single MPI process when specifying CELERITY_DRY_RUN_NODES" or similar.